osbenchmark/benchmark.py

```diff
@@ -139,115 +181,161 @@ def add_workload_source(subparser):
+    )
     create_workload_parser.add_argument(
-        "--workload",
-        required=True,
-        help="Name of the generated workload")
+        "--workload", required=True, help="Name of the generated workload"
+    )
     create_workload_parser.add_argument(
         "--indices",
         type=non_empty_list,
         required=True,
-        help="Comma-separated list of indices to include in the workload")
+        help="Comma-separated list of indices to include in the workload",
+    )
     create_workload_parser.add_argument(
         "--target-hosts",
         default="",
         required=True,
-        help="Comma-separated list of host:port pairs which should be targeted")
+        help="Comma-separated list of host:port pairs which should be targeted",
+    )
     create_workload_parser.add_argument(
         "--client-options",
         default=opts.ClientOptions.DEFAULT_CLIENT_OPTIONS,
-        help=f"Comma-separated list of client options to use. (default: {opts.ClientOptions.DEFAULT_CLIENT_OPTIONS})")
+        help=f"Comma-separated list of client options to use. (default: {opts.ClientOptions.DEFAULT_CLIENT_OPTIONS})",
+    )
     create_workload_parser.add_argument(
         "--output-path",
         default=os.path.join(os.getcwd(), "workloads"),
-        help="Workload output directory (default: workloads/)")
+        help="Workload output directory (default: workloads/)",
+    )
     create_workload_parser.add_argument(
         "--custom-queries",
-        type=argparse.FileType('r'),
-        help="Input JSON file to use containing custom workload queries that override the default match_all query")
+        type=argparse.FileType("r"),
+        help="Input JSON file to use containing custom workload queries that override the default match_all query",
+    )
     create_workload_parser.add_argument(
         "--number-of-docs",
         action=opts.StoreKeyPairAsDict,
-        nargs='+',
+        nargs="+",
         metavar="KEY:VAL",
-        help="Map of index name and integer doc count to extract. Ensure that index name also exists in --indices parameter. " +
-        "To specify several indices and doc counts, use format: <index1>:<doc_count1> <index2>:<doc_count2> ...")
+        help="Map of index name and integer doc count to extract. Ensure that index name also exists in --indices parameter. "
+        + "To specify several indices and doc counts, use format: <index1>:<doc_count1> <index2>:<doc_count2> ...",
+    )
+    create_workload_parser.add_argument(
+        "--concurrent",
+        action="store_true",
+        help="Whether to generate workload concurrently (default: false)",
+    )
+    create_workload_parser.add_argument(
+        "--threads",
+        type=positive_number,
+        default=8,
+        help="Number of threads for dumping documents from indices (default: false)",
+    )
+    create_workload_parser.add_argument(
+        "--bsize",
```

Comment:
Nit: I think it's worth changing this to `--batch-size` to make it easier to understand on first glance and to keep it consistent with every other parameter that spells out the option. One thing we can do in the future is add an abbreviated option for specific parameters. For example, users can either use `--batch-size` or `-b`.